src/pal/inc/pal.h

```diff
@@ -6870,6 +6870,7 @@ class NativeExceptionHolderFactory
     };                                                                          \
     try                                                                         \
     {                                                                           \
+        HardwareExceptionHolder                                                 \
```

Comment:
No, “HardwareExceptionHolder” is a macro that is only active if FEATURE_ENABLE_HARDWARE_EXCEPTIONS is defined and it is defined only for DAC builds.

#ifdef FEATURE_ENABLE_HARDWARE_EXCEPTIONS
#define HardwareExceptionHolder CatchHardwareExceptionHolder __catchHardwareException;
#else
#define HardwareExceptionHolder
#endif // FEATURE_ENABLE_HARDWARE_EXCEPTIONS


From: Jan Vorlicek [mailto:notifications@github.com]
Sent: Wednesday, December 9, 2015 11:49 AM
To: dotnet/coreclr <coreclr@noreply.github.com>
Cc: Mike McLaughlin <mikem@microsoft.com>
Subject: Re: [coreclr] Disable h/w exceptions for coreclr (non-DAC) builds. (#2282)


In src/pal/inc/pal.h<https://github.com/dotnet/coreclr/pull/2282#discussion_r47141882>:

> @@ -6870,6 +6870,7 @@ class NativeExceptionHolderFactory

>      };                                                                          \

>      try                                                                         \

>      {                                                                           \

> +        HardwareExceptionHolder                                                 \

I can see that you have removed the CatchHardwareExceptionHolder base class from the NativeExceptionHolderBase, but you have added it to the macros at all places where we have the native holder. So doesn't the change result the same behavior as before?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/2282/files#r47141882>.